apps/pipelines/nodes/nodes.py

```diff
@@ -907,10 +907,25 @@ def _get_custom_globals(self, state: PipelineState):
                 "get_participant_schedules": participant_data_proxy.get_schedules,
                 "get_temp_state_key": self._get_temp_state_key(state),
                 "set_temp_state_key": self._set_temp_state_key(state),
+                "get_session_state_key": self._get_session_state_key(state["experiment_session"]),
+                "set_session_state_key": self._set_session_state_key(state["experiment_session"]),
             }
         )
         return custom_globals
 
+    def _get_session_state_key(self, session: ExperimentSession):
+        def get_session_state_key(key_name: str):
```

Comment:
We can put various metrics in here as well. Things like the datetime of the last message from the user/ai, the total number of messages etc